### ISSUE TITLE
Plugin Extensions are not detected with ClassLoadingStrategy.APD

### ISSUE THREAD

**Issue by bhanu8784** [pf4j/pf4j#449]
Hi,

I am trying to use pf4j/pf4j: Plugin Framework for Java (PF4J) (github.com) with spring and wanted to use Application first strategy for class loader. 

Issue I am observing is whenever I use ClassLoadingStrategy.APD  then extensions are not identified ( I mean not able to read extensions.idx file from plugin jar). Hence pluginmanager.getExtensions(pluginid) will always return empty.

1.  My application jar (spring boot jar)  also has extensions.idx  which is always getting created  without any extensions as I don't have any extension in my application.
2.  My plugin jar also has extensions.idx with extension classes as expected.

I am setting ClassLoadingStrategy.APD for PluginClassLoader.

My Obervsation:
------------------

When extensionfinder trying to read extensions.idx it will always find one which is in my application.jar and that will not have any extensions. (getResource() method in the PluginClassloader will follow APD while loading the resource as it finds one in parent it will not attempt to load extensions from extensions.idx present in plugin jar)

Thank you
BhanuKiran

**Comment by decebals** [comment]
I can confirm that is a bug but it's somehow expected. 
It's an `extensions.idx` file in app and other ``extensions.idx` in plugin. If `ClassLoadingStrategy.APD` is used, then the `extensions.idx` of app is found first.
I will take a look on the MR proposed by you.

**Comment by alegacy** [comment]
Any progress on this?  I seem to have hit the same issue.   I am seeing a META-INF/extensions.idx that appears to be coming from the pf4j-spring-0.7.0.jar file which is getting used when I use the APD loading strategy.   I am not clear on why that jar file has its own extensions.idx file.

I posted a question on google groups just now:
https://groups.google.com/g/pf4j/c/52MRpvTmMOQ

``` 
2022-10-26 16:17:52.601 DEBUG 60535 --- [           main] org.pf4j.LegacyExtensionFinder           : Reading extensions storages from classpath
2022-10-26 16:17:52.602 DEBUG 60535 --- [           main] org.pf4j.LegacyExtensionFinder           : Read 'file:/Users/foo/pf4j-test-app/build/libs/pf4j-test-app.jar!/BOOT-INF/lib/pf4j-spring-0.7.0.jar!/META-INF/extensions.idx'
```

**Comment by decebals** [comment]
@alegacy I didn't find a good solution yet.
In your case the solution is relative simple because from I see the presence of `META-INF/extensions.idx` in pf4j-sping is useless (no extensions defined). So, we can fix this aspect in pf4j-spring, and that it's all.

**Comment by alegacy** [comment]
@decebals Thanks for confirming.   I have manually removed the `META-INF/extensions.idx` file from the pf4j-spring jar in my gradle cache and confirmed that the plugin extensions are now loaded while using the `ClassLoadingStrategy.APD`.

While we wait for an updated pf4j-spring jar to appear on maven central is there a gradle workaround that could be used to exclude that file automatically either when importing the dependency or when building the bootJar?

**Comment by decebals** [comment]
@alegacy Released pf4j 3.8.0 and pf4j-spring 0.8.0 are now available on Maven Central repository.

**Comment by alegacy** [comment]
Thank you @decebals !

**Comment by etojesaandro** [comment]
I faced the same issue in case if my AppClassLoader contains extension.idx by the same path. It happens as I have several unit-tests in my plugin which emulates the main Application to be started and loads the plugins itself.

**Comment by etojesaandro** [comment]
Created PR to look for extensions meta-info from the plugin first #586

**Comment by decebals** [comment]
Fixed by #586.
